### DISCUSSION TITLE
Community plugin ideas

### DISCUSSION THREAD

**Discussion by craicoverflow** [aerogear/graphback#2179] [Ideas]
Does anyone have any ideas for some community plugins for Graphback? We can guide you through it, or you refer to our [Create your own Graphback plugin](https://graphback.dev/docs/plugins/create) guide!

**Comment by wtrocki** [comment]
explore the creation of the plugin that will provide `token` based pagination. 
Idea here is to change limit and offset and use only token. This will require separate service but I would ignore it for the time being.

**Comment by JWBrownie** [comment]
I want to be able to set a customize type for the ID, for example, a String to which I might insert a validation or custom generator also allow the user to insert IDs, by the way, is that even possible right now?

**Reply by wtrocki** [reply]
There is @id annotation that you can use

**Reply by JWBrownie** [reply]
Yeah, that's good, is there API Documentation on Graphback? I want to know all the available annotations and possibilities to create models, for example, I am working to create a Brickschema Models so some of the fields are displaying complex concatenation of values to represent and ID which by the way I am sure it shouldn't be that way but well that's what they are asking from me. I tried the @id annotation and it works but for autogenerating the id it seems not possible.

**Reply by wtrocki** [reply]
https://graphback.dev/docs/graphql-migrations/db-design

So you can have custom IDs but we do not have auto-generate option if the type is different than ID
